feature/DND-37

Neoenvie opened the pull request.

Neoenvie committed:
DND-37
- added networking model classes
- added Constants.kt needed for request
- added TheDogApi.kt interface with method GET
- added TheDogRepository.kt and TheDogRepositoryImpl.kt with getDogBreeds()
- added NetworkingModule.kt supplying everything TheDogRepositoryImpl and everything needed for it

Neoenvie committed:
DND-37
- added DogFact.kt
- TheDogApi.kt, TheDogRepository.kt, TheDogRepositoryImpl.kt -> added param for function and added function

Neoenvie committed:
DND-37
- added navigation to DogsScreen.kt
- added viewModel for DogsScreen.kt

Neoenvie committed:
DND-37
- added needed implementation in app build.gradle.kts

Neoenvie committed:
DND-37
- added dog photos in DogsScreen.kt

Neoenvie committed:
DND-37
- added DogsUiState.kt and its use
- added Effects.kt reacting to SearchEvent emiting with Toasts

Neoenvie committed:
DND-37 minor changes

Miltentt commented on featureCardBase/src/main/java/com/miltent/featurecardbase/dogs/DogsViewModel.kt:
```diff
@@ -0,0 +1,59 @@
+
+    fun searchDogBreeds() {
+        viewModelScope.launch {
+            val newDogBreedsValue = theDogRepository.getDogBreeds(uiState.value.breedText)
```
Tu też by się przydał try catch, tutaj też może przecież nie przejsć zapytanie z jakiś randomowych powodów

Neoenvie committed:
DND-37
- deleted properties height and weight form Breed
- added Networking versions of Breed BreedFact and Image
- added Mappers for Networking classes
- added UseCases

Neoenvie commented on domain/src/main/java/com/miltent/domain/model/networking/Breed.kt:
```diff
@@ -0,0 +1,19 @@
+import com.miltent.domain.model.networking.Image
+import com.miltent.domain.model.networking.Weight
+
+data class Breed(
```
done

Neoenvie commented on networking/src/main/java/com/miltent/networking/TheDogRepositoryImpl.kt:
```diff
@@ -0,0 +1,33 @@
+class TheDogRepositoryImpl @Inject constructor(
+    val api: TheDogApi,
+    val breedMapper: BreedNetworkingToDomainMapper,
+    val breedFactMapper: BreedFactNetworkingToDomainMapper
```
done

Neoenvie committed:
DND-37
- added bind function for breed mappers
